Migrate project to Antora

vy opened the pull request:
This PR migrates the project to Antora, and provides following benefits:

* Improves navigation issues (lack of ToC, permalink, etc.)
* Addresses UI issues reported in #9
* Uses the same mechanism shared by Logging Parent, Log4Net, Log4j, Log4j Kotlin, Log4j Scala, Log4j Tools, Log4j Transform, etc.

vy committed:
Migrate project to Antora

vy commented:
@ppkarwasz, I'd appreciate your review, and also suggestions on serving assets, which is _complicated_ in Antora. `cyclonedx/vdr.xml`, `doaps/doap_Log4j.rdf`, `xml/ns/*.xsd` files are moved to `/_attachments`. We can use a `.htaccess` or create symlinks in `pom.xml`, etc. What would you suggest?

vy committed:
Add `deploy-site` workflow

ppkarwasz commented on src/site/antora/modules/ROOT/nav.adoc:
```diff
@@ -0,0 +1,29 @@
+* xref:download.adoc[]
+* xref:support.adoc[]
+* xref:security.adoc[]
+* xref:xml/ns/index.adoc[XML Schema]
```
Some links are missing to comply with the [Apache Project Branding Policy](https://www.apache.org/foundation/marks/pmcs#navigation):


```suggestion
* xref:xml/ns/index.adoc[XML Schema]
* https://apache.org[The ASF]
** https://apache.org/licenses/[License]
** https://apache.org/foundation/sponsorship[Donate]
** https://apache.org/foundation/sponsors[Thanks]
```

Whimsy checks do not enforce the `www.` prefix nor the `.html` suffix, so I propose to omit them, since they are pretty much redundant.

vy replied:
Fixed in 79627b35.

ppkarwasz reviewed (changes requested):
Hi @vy, nice work! :100:

This looks very close to ready. I have just a few items to address before merging:

1. **Missing required ASF links**
   The homepage needs the standard Apache project links as detailed in the comment below.

2. **Static files (DOAP, VDR, XSD)**
   These are machine-consumed entry points and their URLs **should** remain stable and canonical (e.g. `https://logging.apache.org/cyclonedx/vdr.xml`). Using Antora attachments changes/obscures those URLs, so I don’t think attachments are the right fit here.
   Using [**UI supplemental static files**](https://docs.antora.org/antora/latest/playbook/ui-supplemental-files/#static-files) seems more appropriate.

3. **Blog not in navigation**
   The blog is currently missing from the nav. We should either add a `blog/index.adoc` page or include the existing blog entries in the navigation.

Once these are addressed, I think this PR should be good to go. Thanks again!

vy committed:
Merge remote-tracking branch 'origin/main' into antora

vy committed:
Move blog to its own page

vy committed:
Add ASF links

vy committed:
Use Antora `supplemental_files` to serve static content

vy committed:
Simplify `antora.tmpl.yml`

vy committed:
Copy static site content using a Maven plugin instead

vy commented:
> 2. **Static files (DOAP, VDR, XSD)**
>    These are machine-consumed entry points and their URLs **should** remain stable and canonical (e.g. `https://logging.apache.org/cyclonedx/vdr.xml`). Using Antora attachments changes/obscures those URLs, so I don’t think attachments are the right fit here.
>    Using [**UI supplemental static files**](https://docs.antora.org/antora/latest/playbook/ui-supplemental-files/#static-files) seems more appropriate.

I had already considered the UI supplemental files route, but refrained from it due to two major drawbacks:

1. You need to explicitly provide every single file in `antora-playbook.yaml > ui.supplemental_files` object
2. You need to change [the UI output directory](https://docs.antora.org/antora/latest/playbook/ui-output-dir/) — I find this a disruptive change (we deviate from Antora defaults) for a very small issue

Nevertheless, I implemented this in 6d30d6e9. Though later on, in e9f718b2, I switched to `maven-resources-plugin` to copy static content, and this doesn't suffer from the aforementioned shortcomings.

> 3. **Blog not in navigation**
>    The blog is currently missing from the nav. We should either add a `blog/index.adoc` page or include the existing blog entries in the navigation.

Fixed in bc1f7c68.

vy commented:
Note that [Activity Monitor](https://logging.apache.org/activity-monitor/) links are removed. I am intending to nuke it in a subsequent PR.

ppkarwasz reviewed (approved):
LGTM, thanks!
